### PR TITLE
feat(settings): add configurable default retry limit (#28944)

### LINKED ISSUE #28944: Add a config option for the default RetryStrategy.Limit
# Summary

Currently, the hardcoded default for RetryStrategy.Limit is 5.

# Motivation

With a large number of app-of-apps, waiting for dependencies can mean the default retry limit of 5 is insufficient.
It would be convenient to set the number of retries in one please, rather than in an override on each and every app.

# Proposal

Read the default limit from a config file.

### PR DESCRIPTION
**Pull request by rithishcodespace**
Fixes #28944

Adds a configurable default retry limit for automated application sync operations through `argocd-cm`.

Currently, when an Application does not specify `spec.syncPolicy.retry`, the Application controller uses a hardcoded retry limit of `5`. This makes it difficult for operators to configure a different default across applications, particularly in app-of-apps deployments.

### Changes

- Add `retry.default.limit` to `argocd-cm`, with a default value of `5`.
- Parse the setting through `SettingsManager`.
- Use the configured value as the default retry limit for automated sync operations.
- Preserve explicit per-Application `spec.syncPolicy.retry` configuration as the higher-priority setting.
- Fall back to `5` when the ConfigMap setting is missing or invalid.
- Document the new configuration option.
- Add unit tests for default, custom, zero, negative, invalid, and per-Application override cases.

### Backward Compatibility

When `retry.default.limit` is not configured, the existing retry limit of `5` is preserved. Application-specific retry configuration continues to take precedence over the global default.

### CLI / UI

No CLI or UI changes are required because this is an `argocd-cm` configuration setting.

### Documentation

Updated `docs/operator-manual/argocd-cm.yaml` to document `retry.default.limit`.

### Testing

- Added settings tests covering default and configured values.
- Added controller tests covering the global default and per-Application override.
- `git diff --check` passes.

Ran successfully:
- `go test ./util/settings/...`
- `go test ./controller/...`
- `make lint`

## Checklist

- [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be discussed with the community — this addresses an existing issue (#28944).
- [x] The title of the PR states what changed and the related issue number.
- [x] I've included "Fixes #28944" in the description.
- [x] The title of the PR conforms to the [Toolchain Guide](https://github.com/argoproj/argo-cd/blob/master/docs/developer-guide/toolchain-guide.md).
- [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them. — N/A, no CLI/UI changes required.
- [x] Does this PR require documentation updates? I've updated the documentation accordingly.
- [x] I've signed off all my commits as required by [DCO](https://github.com/argoproj/argo-cd/blob/master/CONTRIBUTING.md#certificate-of-origin).
- [x] I've written unit and/or e2e tests for my change.
- [ ] All PR checks are green.
- [x] My original issue may be solved by a combination of other changes/features. In this case, I've explained here why I believe the PR is still required.

### CODE REVIEW

File: util/settings/settings.go [hunk 1]
```diff
@@ -1750,6 +1776,14 @@ func updateSettingsFromConfigMap(settings *ArgoCDSettings, argoCDCM *corev1.Conf
 			settings.MaxPodLogsToRender = val
 		}
 	}
+	settings.DefaultRetryLimit = defaultRetryLimit
+	if retryDefaultLimitStr, ok := argoCDCM.Data[settingsRetryDefaultLimitKey]; ok && retryDefaultLimitStr != "" {
+		if val, err := strconv.ParseInt(retryDefaultLimitStr, 10, 64); err != nil {
+			log.Warnf("Failed to parse '%s' key: %v", settingsRetryDefaultLimitKey, err)
```
[the remark this answers is not in the document]
Comment (ppapapetrou76) [reply util/settings/settings.go:1798]: This is a valid comment - please address it

File: util/settings/settings.go [hunk 2]
```diff
@@ -176,6 +176,8 @@ type ArgoCDSettings struct {
 	// RequireOverridePrivilegeForRevisionSync indicates whether giving an external revision during snyc is considered an override.
 	// Up to revision 3.2, this was always false. It is now still false by default, in order to not breaking existing usage.
 	RequireOverridePrivilegeForRevisionSync bool `json:"requireOverridePrivilegeForRevisionSync"`
+	// DefaultRetryLimit defines the global default retry limit for automated application sync operations
+	DefaultRetryLimit int64 `json:"retry.default.limit,omitempty"`
```
Comment (ppapapetrou76) [inline util/settings/settings.go:180]: where is this used?
Why do we need both this and `SettingsManager.GetDefaultRetryLimit()` ?

File: docs/operator-manual/argocd-cm.yaml [hunk 3]
```diff
@@ -382,6 +382,9 @@ data:
   # This is to prevent the UI from becoming unresponsive when rendering a large number of logs. Default is 10.
   server.maxPodLogsToRender: "10"
 
+  # Default retry limit for automated application sync operations when not specified in application spec. Default is 5.
+  retry.default.limit: "5"
```
Comment (ppapapetrou76) [inline docs/operator-manual/argocd-cm.yaml:403]: please use something like `application.sync.retry.limit` to keep the same naming conventions

File: util/settings/settings.go (hunk 2 above)
Comment (rithishcodespace) [reply util/settings/settings.go:180]: Good point. I’ll remove the duplication and use ArgoCDSettings.DefaultRetryLimit directly.

File: docs/operator-manual/argocd-cm.yaml (hunk 3 above)
Comment (rithishcodespace) [reply docs/operator-manual/argocd-cm.yaml:403]: Agreed. I’ll rename it to application.sync.retry.limit to follow the existing naming convention.
